### PR TITLE
Add ScrapeConfig project idea

### PR DESCRIPTION
**Pull request by ArthurSens**
@slashpai, is this the email you want to use? I've copied from the office hour calendar invites :)

### CODE REVIEW

File: mentoring/gsoc/2024/project_ideas.md [hunk 1]
```diff
@@ -26,3 +26,12 @@ If you are a Prometheus-Operator team member and consider mentoring during the G
...
 
 ## Ideas
+
+### Extending ScrapeConfig surface
+
+- Description: The ScrapeConfig is a recently added CRD of Prometheus-Operator. Differently from other monitoring related CRDs like ServiceMonitor and PodMonitor, ScrapeConfig offers a more generic approach by aiming to be a 1:1 equivalent of the Prometheus scrape configuration. Since ScrapeConfig was created pretty recently, it is still far from being a 1:1 mapping of the Prometheus configuration. For us to be able to promote the ScrapeConfig CRD from alpha to beta/stable, we need to cover more missing fields.
+- Recommended Skills: Go, Prometheus
+- Expected project size: small
+- Mentors:
+  - Jayapriya Pai (@slashpai, janantha@redhat.com)
```
Comment (slashpai) [inline mentoring/gsoc/2024/project_ideas.md:36]:
```suggestion
  - Jayapriya Pai (@slashpai, slashpai9@gmail.com)
```

File: mentoring/gsoc/2024/project_ideas.md [hunk 2]
```diff
@@ -26,3 +26,12 @@ If you are a Prometheus-Operator team member and consider mentoring during the G
 ---
 
 ## Ideas
+
+### Extending ScrapeConfig surface
+
+- Description: The ScrapeConfig is a recently added CRD of Prometheus-Operator. Differently from other monitoring related CRDs like ServiceMonitor and PodMonitor, ScrapeConfig offers a more generic approach by aiming to be a 1:1 equivalent of the Prometheus scrape configuration. Since ScrapeConfig was created pretty recently, it is still far from being a 1:1 mapping of the Prometheus configuration. For us to be able to promote the ScrapeConfig CRD from alpha to beta/stable, we need to cover more missing fields.
```
Comment (simonpasquier) [inline mentoring/gsoc/2024/project_ideas.md:32]:
```suggestion
- Description: The ScrapeConfig is a recently added Custom Resource Definition (CRD) of Prometheus-Operator. Differently from other monitoring related CRDs like ServiceMonitor and PodMonitor, ScrapeConfig offers a more generic approach by aiming to be a 1:1 equivalent of the Prometheus scrape configuration. The current state of ScrapeConfig is far from being feature-complete compared to whaat Prometheus configuration offers. For us to be able to promote the ScrapeConfig CRD from alpha to beta/stable, we need to implement more service discoveries as well as missing configuration options.
```

File: mentoring/gsoc/2024/project_ideas.md [hunk 3]
```diff
@@ -26,3 +26,12 @@ If you are a Prometheus-Operator team member and consider mentoring during the G
 ---
 
 ## Ideas
+
+### Extending ScrapeConfig surface
+
+- Description: The ScrapeConfig is a recently added Custom Resource Definition (CRD) of Prometheus-Operator. Differently from other monitoring related CRDs like ServiceMonitor and PodMonitor, ScrapeConfig offers a more generic approach by aiming to be a 1:1 equivalent of the Prometheus scrape configuration. The current state of ScrapeConfig is far from being feature-complete compared to whaat Prometheus configuration offers. For us to be able to promote the ScrapeConfig CRD from alpha to beta/stable, we need to implement more service discoveries as well as missing configuration options.
```
Comment (slashpai) [inline mentoring/gsoc/2024/project_ideas.md:32]:
```suggestion
- Description: The ScrapeConfig is a recently added Custom Resource Definition (CRD) of Prometheus-Operator. Differently from other monitoring related CRDs like ServiceMonitor and PodMonitor, ScrapeConfig offers a more generic approach by aiming to be a 1:1 equivalent of the Prometheus scrape configuration. The current state of ScrapeConfig is far from being feature-complete compared to what Prometheus configuration offers. For us to be able to promote the ScrapeConfig CRD from alpha to beta/stable, we need to implement more service discoveries as well as missing configuration options.
```

**Review by slashpai — APPROVED** [review]
just a small nit, lgtm

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**